pkgs/by-name/di/dingtalk/package.nix

```
@@ -0,0 +1,271 @@
+  freetype,
+  fribidi,
+  gdk-pixbuf,
+  glib,
+  gnome2,
+  gnutls,
+  graphite2,
+  gtk3,
+  harfbuzz,
+  icu63,
+  krb5,
+  libdrm,
+  libgcrypt,
+  libGLU,
+  libglvnd,
+  libidn2,
+  libinput,
+  libjpeg,
+  libpng,
+  libpsl,
+  libpulseaudio,
+  libssh2,
+  libthai,
+  libxcrypt-legacy,
+  libxkbcommon,
+  mesa,
+  mtdev,
+  nghttp2,
+  nspr,
+  nss,
+  openldap,
+  openssl_1_1,
+  pango,
+  pcre2,
+  qt5,
+  rtmpdump,
+  udev,
+  util-linux,
+  xorg,
+}:
+################################################################################
+# Mostly based on dingtalk-bin package from AUR:
+# https://aur.archlinux.org/packages/dingtalk-bin
+################################################################################
+let
+  libraries = [
+    alsa-lib
+    apr
+    aprutil
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    cups
+    curl
+    dbus
+    e2fsprogs
+    fontconfig
+    freetype
+    fribidi
+    gdk-pixbuf
+    glib
+    gnome2.gtkglext
+    gnutls
+    graphite2
+    gtk3
+    harfbuzz
+    icu63
+    krb5
+    libdrm
+    libgcrypt
+    libGLU
+    libglvnd
+    libidn2
+    libinput
+    libjpeg
+    libpng
+    libpsl
+    libpulseaudio
+    libssh2
+    libthai
+    libxcrypt-legacy
+    libxkbcommon
+    mesa
+    mtdev
+    nghttp2
+    nspr
+    nss
+    openldap
+    openssl_1_1
+    pango
+    pcre2
+    qt5.qtbase
+    qt5.qtmultimedia
+    qt5.qtsvg
+    qt5.qtx11extras
+    rtmpdump
+    udev
+    util-linux
+    xorg.libICE
+    xorg.libSM
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXinerama
+    xorg.libXmu
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXt
+    xorg.libXtst
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
+    xorg.xcbutilwm
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "dingtalk";
+  version = "7.6.15.4102301";
+  src = fetchurl {
+    url = "https://dtapp-pub.dingtalk.com/dingtalk-desktop/xc_dingtalk_update/linux_deb/Release/com.alibabainc.dingtalk_7.6.15.4102301_amd64.deb";
+    hash = "sha256-DWRI6y5eXRuAHe90Pm6VK44+tuBxbNLqUmv9OiK8Fm0=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    qt5.wrapQtAppsHook
+    copyDesktopItems
+  ];
+  buildInputs = libraries;
+
+  # We will append QT wrapper args to our own wrapper
+  dontWrapQtApps = true;
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    ar x $src
+    tar xf data.tar.xz
+
+    mv opt/apps/com.alibabainc.dingtalk/files/version version
+    mv opt/apps/com.alibabainc.dingtalk/files/*-Release.* release
+
+    # Cleanup
+    rm -f release/{*.a,*.la,*.prl}
+    rm -f release/dingtalk_crash_report
+    rm -f release/dingtalk_updater
+    rm -f release/libapr*
+    rm -f release/libcrypto.so.*
+    rm -f release/libcurl.so.*
+    rm -f release/libdouble-conversion.so.*
+    rm -f release/libEGL*
+    rm -f release/libfontconfig*
```

```suggestion
    rm release/lib{apr,crypto,curl,double-conversion,EGL,fontconfig}*
```
And so on. You can remove at least 10 libraries per line.